src/main/java/org/ethereum/beacon/discovery/pipeline/Envelope.java

```diff
@@ -6,15 +6,19 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Random;
 import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
 
 /** Container for any kind of objects used in packet-messages-tasks flow */
 public class Envelope {
+
   private final UUID id;
   private final Map<Field<?>, Object> data = new HashMap<>();
 
   public Envelope() {
-    this.id = UUID.randomUUID();
+    Random random = ThreadLocalRandom.current();
+    this.id = new UUID(random.nextLong(), random.nextLong());
```

Comment:
Given we only use this for logging, I wonder if we'd be better off just using `AtomicLong.getAndIncrement` to assign a simple unique identifier.  Could seed it with a large number if we wanted to keep it reasonably long and easier to search for.